### PR TITLE
Modularized backend code

### LINKED ISSUE #308: Backend code needs refactoring
## Description
As a developer,
I need keep different logical parts of my code separate,
so that the code is more readable, manageable and reusable.

### Update [Required]
- [ ]  Separate folder to deal with mongoose scehmas.
- [ ] Separate folder to deal with http endpoints.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
3 hours

### PR DESCRIPTION
**Pull request by f2015537**
### Description
This PR is an attempt to refactor server.js code into different modules so that the logic is abstracted away from the entry file.

Fixes #308 

### Type of Change:
- Code
- Quality Assurance

### How Has This Been Tested?
All routes have been checked rigorously using POSTMAN.

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged 
  

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by f2015537** [comment]
@bethanyr @Tharangi @BaaniLeen @anumehaagrawal looking forward to your review :) . Currently there are 4 commits but there are important logics in each of them. If you want me to squash them let me know!
Also please merge https://github.com/systers/systers.github.io/pull/306 before this one because it might light to issues if done the other way around

**Comment by f2015537** [comment]
Wow what a battle with codacy!

File: config/keys.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  mongoURI: 'mongodb://systers:systers2018@ds235461.mlab.com:35461/programs'
```
Comment (bethanyr) [inline config/keys.js]: can this also be stored in the environment.prod.ts file?

Comment (f2015537) [reply config/keys.js]: @bethanyr thanks for the review :)
The environment.prod.ts file lives on the client side(Angular). The keys.js file lives on the server side(Node). The client side never makes a call to mongodb://systers:systers2018@ds235461.mlab.com:35461/programs
so I don't think it is necessary to add this mongoURI in the environment.prod.ts file.
Plus this mongoURI is being used for development purposes only. The URI that is used for production is https://systers-github-io.herokuapp.com.

**Comment by f2015537** [comment]
@bethanyr please don't merge this just yet. I'm slightly confused about the linting configuration used for this project.

**Comment by f2015537** [comment]
@bethanyr please don't merge this just yet. I'm a bit confused about the linting configuration on the server side and waiting for @BaaniLeen to clear it out. I may modify this PR depending on what @BaaniLeen has to say.

**Pull request merged**